### ISSUE TITLE
How to get collections and buckets names?

### ISSUE THREAD

**Issue by ygworldr** [valeriansaliou/sonic#271]
Is there a way to get a list of collections and buckets directly from Sonic?

My use case is that I want to build buckets based on some data that I might not know in advance. In the search UI, I would like to be able to specify a drop box to search those specific buckets but I do not know what those are so I would need to query Sonic to ask. Does that make sense?

**Comment by valeriansaliou** [comment]
No, unfortunately as per https://github.com/valeriansaliou/sonic/blob/master/PROTOCOL.md#3%EF%B8%8F⃣-sonic-channel-search-mode its not yet possible to enumerate collections & buckets. You'd need to store them in a separate DB.

**Comment by ygworldr** [comment]
Merci. 

Is this a feature you would be willing to add?

Is this something that would make a good PR for me to do? I know nothing of Rust and have almost no time but … :grinning:

**Comment by valeriansaliou** [comment]
That’s something that would be interesting yes, although I also have no time to build it, so open to PRs. Note that unfortunately this would require breaking changes on the Sonic DB format, as you’d need to add an index listing all collections and buckets. So a migration would be needed on existing Sonic installations.
